src/views/TradingCompetition/components/BattleCta/index.tsx

```diff
@@ -0,0 +1,125 @@
+import {
+  Card,
+  CardBody,
+  Flex,
+  LaurelLeftIcon,
+  LaurelRightIcon,
+  Button,
+  CheckmarkCircleIcon,
+  useWalletModal,
+  useModal,
+} from '@pancakeswap-libs/uikit'
+import useAuth from 'hooks/useAuth'
+import useI18n from 'hooks/useI18n'
+import RegisterModal from '../RegisterModal'
+import { Heading2Text } from '../CompetitionHeadingText'
+import { CompetitionProps } from '../../types'
+
+const StyledCard = styled(Card)`
+  display: inline-flex;
+  background: linear-gradient(180deg, #7645d9 0%, #452a7a 100%);
+
+  svg {
+    margin-bottom: 6px;
+    height: 32px;
+    width: auto;
+    fill: ${({ theme }) => theme.colors.warning};
+  }
+`
+
+const StyledButton = styled(Button)`
+  margin: 16px 20px 0;
+  z-index: 200;
+
+  svg {
+    margin: 0 4px 0 0;
+    height: 20px;
+    width: auto;
+    fill: ${({ theme }) => theme.colors.textDisabled};
+  }
+`
+
+const BattleCta: React.FC<CompetitionProps> = ({
+  registered,
+  account,
+  isCompetitionLive,
+  profile,
+  isInitialized,
+  isLoading,
+}) => {
+  const TranslateString = useI18n()
+  const { login, logout } = useAuth()
+  const { onPresentConnectModal } = useWalletModal(login, logout)
+  const [onPresentRegisterModal] = useModal(<RegisterModal profile={profile} />, false)
+
+  const getButtonText = () => {
+    // No wallet connected
+    if (!account) {
+      return TranslateString(999, 'Connect Wallet')
+    }
+
+    // User not registered
+    if (!registered) {
+      return TranslateString(999, 'Register Now!')
+    }
+
+    // User registered but competition has not started
+    if (registered && !isCompetitionLive) {
+      return (
+        <>
+          <CheckmarkCircleIcon /> {TranslateString(999, 'Registered!')}
+        </>
+      )
+    }
+
+    // User registered and competition live
+    if (registered && isCompetitionLive) {
+      return TranslateString(999, 'Trade Now')
+    }
+
+    // May be useful for debugging - if somehow none of the above conditions are met
+    return 'Error'
+  }
+
+  const handleCtaClick = () => {
+    // No wallet connected
+    if (!account) {
+      onPresentConnectModal()
+    }
+
+    // Wallet connected but user not registered
+    if (account && !registered) {
+      onPresentRegisterModal()
+    }
+
+    // Registered and competition is live
+    if (registered && isCompetitionLive) {
+      window.location.href = 'https://exchange.pancakeswap.finance/#/swap'
+    }
+  }
+
+  const isButtonDisabled = () => isLoading || (registered && !isCompetitionLive)
+
+  return (
+    <StyledCard>
+      <CardBody>
+        <Flex flexDirection="column" justifyContent="center" alignItems="center">
+          <Heading2Text>
+            {isCompetitionLive ? TranslateString(999, 'Now Live!') : TranslateString(999, 'Starting Soon')}
+          </Heading2Text>
+          <Flex alignItems="flex-end">
+            <LaurelLeftIcon />
+            <StyledButton disabled={isButtonDisabled()} onClick={() => handleCtaClick()}>
```

Comment:
This button is also disabled when a user profile is still loading (as the profile data is needed when determining which registration modal to present). I toyed with the idea of adding 

```js
  endIcon={isLoading ? <AutoRenewIcon spin color="currentColor" /> : null}
```

But it looked a bit weird. Can add if we think it would be a good addition.

Reply:
Nah